hubblestack/extmods/hubble_mods/ssl_certificate.py

```diff
@@ -0,0 +1,315 @@
+    return cert
+
+
+def _get_cert_from_endpoint(server, port=443, ssl_timeout=3):
+    try:
+        log.debug("ssl_certificate is checking for ssl cert on {0}:{1}".format(server, port))
+        hostport = (str(server), int(port))
+        setdefaulttimeout(ssl_timeout)
+        cert_details = ssl.get_server_certificate(hostport)
+    except Exception as e:
+        log.error('Unable to retrieve certificate from {0}. Error: {1}'.format(server, e))
+        cert_details = None
+    if not cert_details:
+        return None
+    return cert_details
+
+
+def _get_cert_from_file(cert_file_path):
+    try:
+        log.debug("ssl_certificate is checking for ssl cert from path {0}".format(cert_file_path))
+        with open(cert_file_path) as cert_file:
+            cert_details = cert_file.read()
+    except IOError as e:
+        log.error('File not found: {0}. Error: {1}'.format(cert_file_path, e))
+        return None
+    return cert_details
+
+
+def _parse_cert(cert, host, port, path):
+    """
+    load the certificate using OpenSSL and parse needed params.
+    """
+    log.debug("Parsing the fetched certificate")
+    cert_details = {}
+    try:
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        cert_details['ssl_src_port'] = str(port)
+        cert_details['ssl_src_host'] = str(host)
+        cert_details['ssl_src_path'] = str(path)
+        if x509.get_issuer():
+            issuer_components = _format_components(x509.get_issuer())
+            cert_details['ssl_issuer_common_name'] = issuer_components.get('CN', "None")
+        if x509.get_subject():
+            subject_components = _format_components(x509.get_subject())
+            cert_details['ssl_subject_country'] = subject_components.get('C', "None")
+            cert_details['ssl_subject_organisation'] = subject_components.get('O', "None")
+            cert_details['ssl_subject_organisation_unit'] = subject_components.get('OU', "None")
+            cert_details['ssl_subject_common_name'] = subject_components.get('CN', "None")
+        not_after = datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%SZ")
+        not_before = datetime.strptime(x509.get_notBefore().decode('utf-8'), "%Y%m%d%H%M%SZ")
+        has_expired = x509.has_expired()
+        cert_details['ssl_cert_version'] = str(x509.get_version())
+        cert_details['ssl_has_expired'] = True if has_expired == 1 else False
+        cert_details['ssl_serial_number'] = str(x509.get_serial_number())
+        cert_details['ssl_end_time'] = str(not_after)
+        cert_details['ssl_start_time'] = str(not_before)
+        cert_details['ssl_signature_algorithm'] = str(x509.get_signature_algorithm())
+        cert_details['ssl_cert_pem'] = str(cert)
+        cert_details['ssl_subject_alternative_names'] = _get_certificate_san(x509)
+    except Exception as e:
+        cert_details['error'] = "An error occurred while parsing certificate - {0}".format(e)
```

Comment:
In case of exception, this is silently passing.
In this code, nobody is checking for 'error' result. Please check, if this is ok.

Reply:
fixed it in the caller

Reply:
done